### ISSUE TITLE
Computer/ Microcontroller not closing TCP sockets properly during shut down.

### ISSUE THREAD

**Issue by rjsmith2** [MightyPirates/OpenComputers#1346]
I noticed that they're not shutting down their TCP connections upon power down. Any computer with listening TCP sockets (through internet.open or internet.connect for microcontroller) and while shutting them down by using power button, they do nothing and still keep their TCP connections alive until its powered up again.  Expected result is that they terminate all TCP connections, but instead they're still running even tho its not powered anymore.

You could kill TCP by closing them using programs and then shut down. But for microcontroller, you can not execute any programs by hand before shutting down. Their TCP connection will probably be alive until server reboots or timeout after a while.

I am attaching a Lua script for computers to help reproduce the bug: http://pastebin.com/M9T2iHcv

**Comment by gjgfuj** [comment]
That's because they are opened via the card, and stay attached to the card.
so if the computer calling it shuts down, oh well. I'd hope that if it ran
out of power the connection would be terminated though.

**Comment by fnuecke** [comment]
The connections actually already [should get closed when the owning computer powers down for any reason](https://github.com/MightyPirates/OpenComputers/blob/master-MC1.7.10/src/main/scala/li/cil/oc/server/component/InternetCard.scala#L105-L109). If you're absolutely sure this is the case, and they _are_ being reset when powering it back on, that'd suggest the `computer.stopped` message doesn't arrive or something... I'll look into it.

**Comment by rjsmith2** [comment]
gjgfuj,  that is why I'm issuing this report. Like what fnuecke said, it should be terminating the connections when powered down, but does not as of version 1.5.13.28.

I'm not sure what else to say.  They do not terminate their connections at all when shutting down. I have those computers relaying to servers through my nodejs server. Nodejs will log any terminated connections to my console so that way I know when its terminated. This is on single player, by the way.

 When I shut computers down (their power light goes away,) the nodeJS isn't getting any termination signal from them. The nodejs can still send something to it but computer won't process it at all.  

I noticed another things: First,  I destroyed the unpowered computer that is keeping the socket connection alive, it STILL wouldn't terminate the connection. Second, it also still would not kill the connection if I quit to main menu. It only send termination if I quit the game entirely. Third, if I quit to main menu and back into my game back and forth, it still won't send a termination signal. Could this be the sign of memory leak?

The only way I know the computers WILL terminate their signal if:
1. If i kill the single game (to desktop, not main menu)
2. Make a script to close the socket manually. 
3. Rebooting the computer.
